frontend/src/metabase/components/TextInput.jsx

```diff
@@ -1,19 +1,42 @@
 import React, { forwardRef } from "react";
 import PropTypes from "prop-types";
-import styled, { css } from "styled-components";
-import { color } from "metabase/lib/colors";
 import { t } from "ttag";
 
 import Icon from "metabase/components/Icon";
 
+TextInput.propTypes = {
+  onChange: PropTypes.func.isRequired,
+  placeholder: PropTypes.string,
+  value: PropTypes.string,
+  type: PropTypes.string,
+  autoFocus: PropTypes.bool,
+  className: PropTypes.string,
+  hasClearButton: PropTypes.bool,
+  icon: PropTypes.node,
+  padding: PropTypes.oneOf(["sm", "md"]),
+  borderRadius: PropTypes.oneOf(["sm", "md"]),
+  variant: PropTypes.oneOf(["main", "admin"]),
+};
+
+import {
+  TextInputRoot,
+  ClearButton,
+  IconWrapper,
+  Input,
+} from "./TextInput.styled";
+
```

Comment:
I suggest we always put imports before any other statements. I guess it's a widespread rule not only for JavaScript but for many other programming languages. Does it make sense?

Reply:
Oh, thanks for noticing, this was not intentional